commit f87063901afca4528612a2e2037f2388c9e70e5d
Author: Rui Loura <rui@joyent.com>
Date:   2019-07-12T12:41:10-04:00 (3 months ago)
    
    MANTA-4409 Arbitrary MantaObject in rust-libmanta should format manta_storage_id as domain name

diff --git a/Cargo.toml b/Cargo.toml
index 63dea15..c7e6354 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -12,3 +12,4 @@ uuid = {version = "0.7.4", features = ["v4"] }
 md5 = "0.6.1"
 base64 = "0.10.1"
 rand = "0.6.5"
+regex = "1.1.9"
diff --git a/src/lib.rs b/src/lib.rs
index 717268f..e112549 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -1,13 +1,4 @@
-/*
- * Copyright 2019 Joyent, Inc.
- */
+// Copyright 2019 Joyent, Inc.
+
 pub mod moray;
 mod util;
-
-#[cfg(test)]
-mod tests {
-    #[test]
-    fn it_works() {
-        ()
-    }
-}
diff --git a/src/moray.rs b/src/moray.rs
index ebf0098..c138315 100644
--- a/src/moray.rs
+++ b/src/moray.rs
@@ -1,6 +1,4 @@
-/*
- * Copyright 2019 Joyent, Inc.
- */
+// Copyright 2019 Joyent, Inc.
 
 use crate::util;
 use base64;
@@ -75,10 +73,16 @@ pub struct MantaDirectory {
 // Implement Arbitrary traits for testing
 impl Arbitrary for MantaObjectShark {
     fn arbitrary<G: Gen>(g: &mut G) -> MantaObjectShark {
-        let len = g.gen::<u8>() as usize;
+        let len = g.gen_range(1, 100) as usize;
+        let msid = format!(
+            "{}.{}.{}",
+            len,
+            util::random_string(g, len),
+            util::random_string(g, len)
+        );
         MantaObjectShark {
             datacenter: util::random_string(g, len),
-            manta_storage_id: util::random_string(g, len),
+            manta_storage_id: msid,
         }
     }
 }
@@ -144,3 +148,37 @@ impl Arbitrary for MantaObject {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use quickcheck::quickcheck;
+    use regex::Regex;
+    use std::str::FromStr;
+
+    quickcheck!(
+        fn create_manta_object(mobj: MantaObject) -> bool {
+            dbg!(&mobj);
+
+            let str_etag = Uuid::from_str(mobj.etag.as_str());
+            let str_owner = Uuid::from_str(mobj.owner.as_str());
+            let str_object_id = Uuid::from_str(mobj.object_id.as_str());
+            assert!(str_etag.is_ok());
+            assert!(str_owner.is_ok());
+            assert!(str_object_id.is_ok());
+
+            assert_eq!(str_etag.unwrap().to_string(), mobj.etag);
+            assert_eq!(str_owner.unwrap().to_string(), mobj.owner);
+            assert_eq!(str_object_id.unwrap().to_string(), mobj.object_id);
+
+            let re = Regex::new(r"(?i)\d+.[a-z0-9-]+.[a-z0-9-]+").unwrap();
+
+            for shark in mobj.sharks.iter() {
+                dbg!(&shark.manta_storage_id);
+                assert!(re.is_match(&shark.manta_storage_id));
+            }
+
+            true
+        }
+    );
+}
diff --git a/src/util.rs b/src/util.rs
index 64760d2..5b6bd0d 100644
--- a/src/util.rs
+++ b/src/util.rs
@@ -1,6 +1,4 @@
-/*
- * Copyright 2019 Joyent, Inc.
- */
+// Copyright 2019 Joyent, Inc.
 
 use quickcheck::Gen;
 use rand::distributions::Alphanumeric;
